flink/src/main/java/org/apache/zeppelin/flink/FlinkSqlInterrpeter.java

```diff
@@ -0,0 +1,227 @@
+  }
+
+  @Override
+  public void open() throws InterpreterException {
+    flinkInterpreter =
+            getInterpreterInTheSameSessionByClassName(FlinkInterpreter.class);
+  }
+
+  @Override
+  public InterpreterResult interpret(String st,
+                                     InterpreterContext context) throws InterpreterException {
+    LOGGER.debug("Interpret code: " + st);
+    flinkInterpreter.getZeppelinContext().setInterpreterContext(context);
+    flinkInterpreter.getZeppelinContext().setNoteGui(context.getNoteGui());
+    flinkInterpreter.getZeppelinContext().setGui(context.getGui());
+
+    checkLocalProperties(context.getLocalProperties());
+
+    // set ClassLoader of current Thread to be the ClassLoader of Flink scala-shell,
+    // otherwise codegen will fail to find classes defined in scala-shell
+    ClassLoader originClassLoader = Thread.currentThread().getContextClassLoader();
+    try {
+      Thread.currentThread().setContextClassLoader(flinkInterpreter.getFlinkScalaShellLoader());
+      return runSqlList(st, context);
+    } finally {
+      Thread.currentThread().setContextClassLoader(originClassLoader);
+    }
+  }
+
+
+  protected abstract void checkLocalProperties(Map<String, String> localProperties)
+          throws InterpreterException;
+
+  private Optional<SqlCommandParser.SqlCommandCall> parse(String stmt) {
+    // normalize
+    stmt = stmt.trim();
+    // remove ';' at the end
+    if (stmt.endsWith(";")) {
+      stmt = stmt.substring(0, stmt.length() - 1).trim();
+    }
+
+    // parse
+    for (SqlCommandParser.SqlCommand cmd : SqlCommandParser.SqlCommand.values()) {
+      final Matcher matcher = cmd.pattern.matcher(stmt);
+      if (matcher.matches()) {
+        final String[] groups = new String[matcher.groupCount()];
+        for (int i = 0; i < groups.length; i++) {
+          groups[i] = matcher.group(i + 1);
+        }
+        return cmd.operandConverter.apply(groups)
+                .map((operands) -> new SqlCommandParser.SqlCommandCall(cmd, operands));
+      }
+    }
+    return Optional.empty();
+  }
+
+  private InterpreterResult runSqlList(String sql, InterpreterContext context) {
+    List<SqlInfo> sqlLists = SqlLists.getSQLList(sql);
+    List<SqlCommandParser.SqlCommandCall> sqlCommands = new ArrayList<>();
+    for (SqlInfo sqlInfo : sqlLists) {
+      Optional<SqlCommandParser.SqlCommandCall> sqlCommand = parse(sqlInfo.getSqlContent());
+      if (!sqlCommand.isPresent()) {
+        return new InterpreterResult(InterpreterResult.Code.ERROR, "Invalid Sql statement: "
+                + sqlInfo.getSqlContent());
+      }
+      sqlCommands.add(sqlCommand.get());
+    }
+    for (SqlCommandParser.SqlCommandCall sqlCommand : sqlCommands) {
+      try {
+        callCommand(sqlCommand, context);
+        context.out.flush();
+      }  catch (Throwable e) {
+        LOGGER.error("Fail to run sql:" + sqlCommand.operands[0] + "\n"
+                + ExceptionUtils.getStackTrace(e));
```

Comment:
why not `LOGGER.error("Fail to run sql:" + sqlCommand.operands[0], e);`?

Reply:
fixed